apps/mark-scan/frontend/src/lib/gamepad.ts

```diff
@@ -1,16 +1,17 @@
 import { Button } from 'react-gamepad';
 import { mod } from '../utils/mod';
 
+enum QuerySelector {
+  Audio = 'button:not([aria-hidden="true"]):not([disabled])',
+  PAT = 'button:not([aria-hidden="true"]):not([disabled]):not([tabindex="-1"]), [role="button"]:not([aria-hidden="true"]):not([disabled]):not([tabindex="-1"])',
```

Comment:
@jonahkagan @kofi-q any thoughts on separating out these query selectors? With audio focus logic still in flux I don't want to step on any toes, but I do think our ultimate goal should be to unite screen reader and PAT focus behavior.

Reply:
Hmm, weird that `role="button"` didn't work. Oh well

I'm not familiar enough with our screen reader to know the answer to your second question. @kofi-q do you have any ideas?

It's also fine to just have both for now if we're not sure

Reply:
Missed these threads earlier, but also don't have too much historical context on the existing gamepad implementation -- my understanding is that it's essentially trying to simulate browser `tab`/`shitft+tab` behaviour for the up/down buttons. I did wonder if we would be better off mapping the gamepad to keyboard `tab`/`shift+tab` key presses at the system level instead, which would save us from re-implementing this browser functionality.

That being said, we only expect UI controls like buttons and options to be focusable for audio readouts (we do have an initial focus on an introductory text div with `id="audiofocus"` on page load, but that's separate from the accessible controller navigation). That makes me think uniting the focus selectors should be fine/is probably what we want.

(Also, how much control do we have over the PAT -> keyboard mapping? Maybe we can map the focus control to `tab` key instead?)

Reply:
> Also, how much control do we have over the PAT -> keyboard mapping? Maybe we can map the focus control to `tab` key instead?

I believe we can simulate any keypress we want, so `tab` and `shift+tab` should work fine. I'm not sure whether @kshen0 had a reason to use something else, though.

Reply:
That's a good point. Here's what we could support if we send `tab`, `shift+tab`, and `enter` directly from the code where we handle input signals:

1. All behavior for the built-in PAT input
2. Some behavior for the accessible controller. For example, current left and right button behavior would be unsupported because they must rely on querying the DOM
3. No behavior for the USB PAT input switch, used in development only

(3) is a luxury but (2) is needed for production, so we'd need this gamepad abstraction in some form no matter what. Given that, I think it's cleaner to keep all focus behavior in one place - thoughts?

Reply:
I'm thinking the ideal end state is that we no longer need `gamepad.ts` if accessible controllers can map their "focus next/previous" controls to `tab`/`shift+tab` and their "next/previous screen" controls to something like `ArrowLeft/ArrowRight`, then relevant screens can implement handlers for the left/right keys as needed.

After reading through the code, the expectation that a screen will render buttons with `id="previous"`/`id="next"` to get left/right working feels a little brittle. And I noticed the current gamepad behaviour is broken for contest cards after the recent `tabindex=-1` change (it focuses on the "change" buttons instead of the full contest card) - that's what worries me a bit about trying to re-implement browser focus functionality.

Even if we do want to move in the direction of relying on `tab`/`shift+tab`, I think it's fine to add to `gamepad.ts` for now (unifying the query selectors) and then refactoring as a whole later as a separate project.

Reply:
Agreed offline that sending `tab`, `shift+tab`, and `enter` from lower-level device signal handlers is better. We can reduce `gamepad.ts` down to handling just the USB PAT input switch (used in development only).

Reply:
[9f6b023](https://github.com/votingworks/vxsuite/pull/4284/commits/9f6b0236a04e9335159cbf2649f00c1ff1a390cf)